src/cell.rs

```diff
@@ -22,8 +22,8 @@ pub struct Cell {
 impl Cell {
     /// Create a new `Cell` initialized with content from `string`.
     /// Text alignment in cell is configurable with the `align` argument
-    pub fn new_align(string: &str, align: Alignment) -> Cell {
-        let content: Vec<String> = string.lines().map(|x| x.to_string()).collect();
+    pub fn new_align<S: AsRef<str>>(string: S, align: Alignment) -> Cell {
```

Comment:
The thing is, neither `AsRef<str>` family of types nor `ToString` one is subset of the other one. But considering:
- `AsRef<str>` is for types already representable as a string without additional converting;
- `ToString` is to create a string representation of an instance of the type;

we can conclude that every `AsRef<str>` type can trivially implement `ToString`, so in fact `ToString` is broader and more backwards-compatible. Me being unable to give an example of `T: AsRef<str> + !ToString` only gives a point to this idea.